Lib/test/test_abc.py

```diff
@@ -410,6 +410,36 @@ class C:
             with self.assertRaises(TypeError):
                 issubclass(C(), A)
 
+            # bpo-34441: Check that issubclass() doesn't crash on bogus
+            # classes.
+            bogus_subclasses = [
+                None,
+                lambda x: [],
+                lambda: 42,
+                lambda: [42],
+            ]
+
+            for i, func in enumerate(bogus_subclasses):
+                class S(metaclass=abc_ABCMeta):
+                    __subclasses__ = func
+
+                with self.subTest(i=i), \
+                        self.assertRaises(TypeError):
```

Comment:
This is mostly a cosmetic change, I find the following style more readable (and we usually try to avoid using a backslash if it's possible to write a readable code):

```py
with self.subTest(i=i):
    self.assertRaises(TypeError, issubclass, int, S)
```

Or:

```py
with self.subTest(i=i):
    with self.assertRaises(TypeError):
        issubclass(int, S)
```

Reply:
OK, I went for the second one for consistency with other checks.